Create Task Form Changes

AaronDzf opened the pull request:
•Moved create task form from a separate php file into modals on the task list view
•Fixed a few typos
•Changed variable names in SQL query to accommodate for new environment

• Backlogged task from sprint 2

Test for various user input parameters:
Task title, Task description, Effort(integer), Type of work, Due date

AaronDzf committed:
Moved create task form from a separate html file into modals on the task
list view
Fixed a few typos
Changed variable names in SQL query to accomodate for new environment

Test for various user input parameters:
Task title, Task description, Effort(integer), Type of work, Due
date

gaurav613 commented on listView.php:
```diff
@@ -65,6 +65,11 @@
     Edit List Title
     </button>
 
+    <!-- button to edit title of an existing list -->
```
For some reason, the board page is displaying two 'Add new task' buttons - one which links to the modal, and another which links to the full page.

Looks like you forgot to remove the older code - should be a quick and straightforward fix.

gaurav613 replied:
Also, the comment is misleading - mentions 'button to edit list' when it's to add a new task.

AaronDzf replied:
Will make changes - thanks.

gaurav613 commented on listView.php:
```diff
@@ -351,6 +400,68 @@
+              <label for="addTaskTypeWork"> Type of Work</label>
+              <input style="height:50px;width:160px;font-size:14pt;"type="text" class="form-control" name="addTaskTypeWork">
+              <small id="" class="form-text text-muted">Detail of work type</small>
+            </div>
```
Performed Unit Testing for the following cases(Effort Level and Type of Work):

- Null Inputs: The task doesn't get created, but there is no message prompting the user to add any input.
To fix this, I think there should be a `required` tag that can be added to make sure this doesn't happen, or add some default values in, just like for due date input

- Negative Input: It's accepting negative values for some reason, and also displaying it within the task attributes.
To fix this, I think there should be a lower bound(of 0) added to the inputs.

- Decimal Input: It's accepting decimal values as inputs too.
To fix this, I think the form can be made to only allow `int` datatypes as input, and display an error if not so.

Overall, I believe that for these inputs, rather than having free-range values, the user can be given different 'levels' as options(either in the form of radio buttons, check-boxes, or dropdown lists). This hasn't been discussed in the story though, so it isn't a requirement as of now. However, this can help to mitigate any chances of edge case errors as described above.

AaronDzf replied:
• Null inputs: I will look into a default value. By design, a task will only be created when a task title and task list has been selected.
• Negative input: I agree with the suggested changes
• Decimal input: I agree with the suggested changes

gaurav613 commented on listView.php:
```diff
@@ -351,6 +400,68 @@
+            <label for="addListID"> Choose Task List</label>
+              <br>
+              <select name = "addListID">
+                  <option value="" disabled selected>Select Task List</option>
```
Tested this for null input(didn't select any option), and the task didn't get created. To fix this, either a `required` element can be added to the form, or the default selection can be some existing list.

AaronDzf replied:
Yes, by design a task cannot be added without choosing a corresponding task list. I will look into adding a required element.

gaurav613 reviewed (changes requested):
The UI looks good - the modal has been implemented well. Just one change to be made regarding the repeating button, as commented below.

Also, there are still some issues that I faced regarding the functionality of this, which I found when performing unit tests for the cases mentioned below. This PR isn't focussing on the functionality, so these changes aren't required(I think they would be related to the PR for Story 30 - Task Creation). But since we received feedback to improve our testing from the previous sprint, I feel like it was important to look into it.

Otherwise, everything else looks good.

gaurav613 commented:
I am approving this for now, as the main goal(implementing the modal) has been achieved- it can be reviewed again after the functionality changes are made, and then be merged.

**Don't merge until the UI changes are made**

AaronDzf committed:
Made revisions based on review
- Changed comment on line 68
- Effort level has been fixed to an integer between 0 and 10, form will
  not accept any values out of that range
- Requirement validation added for task title and task list
- Performed other unit tests including:
- Task title will accept any varchar up to a max of 255 characters
- Task description will accept any varchar up to a max of 255 characters
- Effort is fixed integer between 0 and 10
- Type of work will accept any varchar up to a max of 255 characters
- Due date, WIP, waiting to check if calendar API will be used

gaurav613 commented:
Thanks for adding in these changes! The functionality should work well, even for edge cases now. 👍
